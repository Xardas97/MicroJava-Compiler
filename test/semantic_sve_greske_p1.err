ERROR 22:42:25,318 - Linija 2: Konstanti greska_int je dodeljen izraz pogresnog tipa
ERROR 22:42:25,319 - Linija 3: Konstanti greska_char je dodeljen izraz pogresnog tipa
ERROR 22:42:25,319 - Linija 9: Nepostojeci tip: imt
ERROR 22:42:25,320 - Linija 12: Identifikator nepostojeci ne postoji
ERROR 22:42:25,320 - Linija 12: Dodela vrednosti nije moguca, tipovi nisu kompatibilni
ERROR 22:42:25,321 - Linija 13: Dodela vrednosti nije moguca, tipovi nisu kompatibilni
ERROR 22:42:25,321 - Linija 15: Identifikator nepostojeci ne postoji
ERROR 22:42:25,321 - Linija 15: Samo celobrojne vrednosti se mogu inkrementirati ili dekrementirati
ERROR 22:42:25,321 - Linija 18: Identifikator i nije funkcija
ERROR 22:42:25,321 - Linija 20: Break se moze koristiti samo u do-while blokovima
ERROR 22:42:25,321 - Linija 21: Continue se moze koristiti samo u do-while blokovima
ERROR 22:42:25,321 - Linija 23: Ne moze se raditi ispisivanje niza
ERROR 22:42:25,322 - Linija 11: Funkciji method1 fali return naredba koja je van grana
ERROR 22:42:25,322 - Linija 29: Operacije AND i OR i uslovi se koriste samo sa bool vrednostima
ERROR 22:42:25,323 - Linija 31: Pogresan broj parametara funkcije method1
ERROR 22:42:25,323 - Linija 31: Parametar #1 poziva funkcije method1 je pogresnog tipa
ERROR 22:42:25,323 - Linija 33: Relacione operacije se ne mogu koristiti sa nekompatibilnim tipovima
ERROR 22:42:25,324 - Linija 34: Mogu se negirati samo celobrojne vrednosti
ERROR 22:42:25,324 - Linija 34: Relacione operacije se ne mogu koristiti sa nekompatibilnim tipovima
ERROR 22:42:25,324 - Linija 35: Povratna vrednost je pogrešnog tipa
ERROR 22:42:25,325 - Linija 37: Operacija % se moze koristiti samo sa celobrojnim vrednostima
ERROR 22:42:25,325 - Linija 38: Velicina niza u strukturi mora biti celobrojna vrednost
ERROR 22:42:25,325 - Linija 42: Return se ne moze koristiti u switch blokovima
ERROR 22:42:25,326 - Linija 42: Return izraz ove funkcije mora imati povratnu vrednost
ERROR 22:42:25,326 - Linija 40: Default bloku fali yield naredba
ERROR 22:42:25,326 - Linija 40: Dodela vrednosti nije moguca, tipovi nisu kompatibilni
ERROR 22:42:25,326 - Linija 47: Svi blokovi switch-a moraju da vracaju isti tip rezultata
ERROR 22:42:25,326 - Linija 26: Funkciji method2 fali return naredba koja je van grana
ERROR 22:42:25,326 - Linija 1: Mora postojati ulazna metoda 'main'
ERROR 22:42:25,327 - Parsiranje NIJE uspesno zavrseno!
