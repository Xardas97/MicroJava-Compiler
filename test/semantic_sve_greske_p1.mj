program semantic_sve_grekse_p1
    const int greska_int = false;
    const char greska_char = 3;

    int i;
    char c;
    int ar[];

    imt x;
{
    int method1(int c) {
        nepostojeci = 3;
        i = true;

        nepostojeci++;
        c--;

        i();
        
        break;
        continue;
        
        print(ar);
    }

    int method2() {
        do {

        } while (1);
    
        method1(c, i);
        
        if (i > c)
            if (i != -c)
                return true;

        i = i % c;
        ar = new int [c];
        
        c = switch(i) {
            case 0: yield 0;
            default: return;
        };

        c = switch(i) {
            case 0: yield 'a';
            default: yield 1;
        };
    }
}