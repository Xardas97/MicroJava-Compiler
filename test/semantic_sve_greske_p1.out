INFO  22:42:25,319 - Linija 5: Nadjeno Var i: Int, 0, 0
INFO  22:42:25,319 - Linija 6: Nadjeno Var c: Char, 1, 0
INFO  22:42:25,319 - Linija 7: Nadjeno Var ar: Array of Int, 2, 0
INFO  22:42:25,319 - Linija 9: Nadjeno Var x: None, 3, 0
INFO  22:42:25,320 - Linija 11: Nadjeno Meth method1: Int, 0, 0
INFO  22:42:25,320 - Linija 11: Nadjeno Var c: Int, 0, 1
INFO  22:42:25,320 - Linija 12: Koriscenje argumenta Var noObj: None, -1, -1
INFO  22:42:25,320 - Linija 13: Koriscenje argumenta Var i: Int, 0, 0
INFO  22:42:25,321 - Linija 15: Koriscenje argumenta Var noObj: None, -1, -1
INFO  22:42:25,321 - Linija 16: Koriscenje argumenta Var c: Int, 0, 1
INFO  22:42:25,321 - Linija 18: Koriscenje argumenta Var i: Int, 0, 0
INFO  22:42:25,322 - Linija 26: Nadjeno Meth method2: Int, 0, 0
INFO  22:42:25,323 - Linija 31: Poziv Meth method1: Int, 0, 1
INFO  22:42:25,326 - ==============SADRZAJ TABELE SIMBOLA===================
INFO  22:42:25,327 - 
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: bool, -1, -1 
Prog semantic_sve_grekse_p1: notype, 0, 1 
   Var i: int, 0, 0 
   Var c: char, 1, 0 
   Var ar: Arr of int, 2, 0 
   Var x: notype, 3, 0 
   Meth method1: int, 0, 1 
      Var c: int, 0, 1 

   Meth method2: int, 0, 0 



